### PR TITLE
feat(lexer): support multi-line comments

### LINKED ISSUE #232: Support multiline comments
Currently we only support single-line comments i.e. `// this is a comment`.

Cypher also supports multi-line style using, let support this as well:
```
/* 
This is a multi-line comment
Woot
*/
```

### PR DESCRIPTION
**Pull request by naorpeled**
## Summary

Adds support for Cypher-style multi-line comments `/* ... */` alongside the existing single-line `//` comments, across the backend lexer, the web editor's query preprocessing, and syntax highlighting in both the web editor and the interactive shell.

Closes #232

```graphql
/*
This is a multi-line comment
Woot
*/
MATCH (d:Deployment) RETURN d.spec.replicas
```

## Backend lexer (`pkg/core/lexer.go`)

Extends the comment branch in `NextToken()`. On seeing `/`, the lexer switches on the next character: `/` → single-line (unchanged), `*` → **new** multi-line skip until `*/` (spanning newlines), else → `ILLEGAL`. The closer is detected by consuming each rune then checking `Peek()` for `/`, which makes the boundaries correct:

- **`/*/` is unterminated** — the `/` after `*` is comment body.
- **Comments do not nest** — first `*/` closes the comment (openCypher semantics).
- **Unterminated comments surface as `ILLEGAL "/*"`** so the parser reports a syntax error.

## Web query preprocessing (`web/src/utils/queries.ts`)

The web editor runs `;`-separated queries, flattening text and splitting. The comment handling is a string-aware state machine (mirroring the lexer): comment markers and `;` inside string literals are ignored (`"http://x"`, `"a;b"`), comments act as token separators, an unterminated `/*` is kept verbatim and ends splitting so the backend surfaces the error, and empty queries are filtered out.

## Syntax highlighting

- **Web** (`QueryInput.tsx`, `web/src/utils/cypherHighlight.ts`): the Prism cypher grammar only matched `//`. Importing react-syntax-highlighter pulls in the full Prism build, which registers the stock cypher grammar into the shared refractor instance first, and `refractor.register()` won't overwrite it — so registering an extended grammar is a no-op. Instead the comment rule is patched directly on the live grammar to also match `/* */`.
- **Shell** (`cmd/cyphernetes/shell.go`): `Paint()` splits each line into code/comment segments, dims `//` and `/* */` comments (grey), and runs the existing colorizer only on code. String literals are skipped while scanning so markers inside them (e.g. `"http://x"`) aren't treated as comments.

## Tests

- `TestLexerComments` (13), `TestParseQueryWithComments` (AST-equivalence + error case).
- `splitQueries` unit tests (14): comments, strings containing comment/`;` markers, token-boundary preservation, unterminated comments, trailing-`;` whitespace, comment-only input.
- `cypherHighlight` unit tests + a render test that mounts the real highlighter (no-wrap and `wrapLines`) and asserts `/* */` produces comment tokens.
- `TestSyntaxHighlighterComments` for the shell painter; existing `TestSyntaxHighlighterPaint` unchanged.
- Docs updated in `docs/docs/language.md`.

All test-first (RED → GREEN). `pkg/core` + `cmd/cyphernetes` Go suites pass, web `tsc --noEmit` clean, web production build succeeds.

> Note: the embedded `cmd/cyphernetes/web/` bundle is regenerated at build/release time, so it's not in this diff (source-only).

### CODE REVIEW

File: web/src/utils/queries.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    // Remove single-line comments (up to the end of the line).
+    .replaceAll(/\/\/.*\n/g, '')
+    // Flatten remaining newlines into spaces.
+    .replace(/\n/g, ' ');
+
+  return normalized
+    .replace(/;$/, '')
+    .split(';')
+    .map((q) => q.trim());
+}
```
[the remark this answers is not in the document]
Comment (naorpeled) [reply web/src/utils/queries.ts:82]: Addressed in 1938034. Replaced the regex stripping in `splitQueries()` with a string-aware state machine: comment markers and `;` inside string literals are ignored (e.g. `"http://x"`, `"a;b"`), comments are replaced with a space so token boundaries are preserved (`MAT/*x*/CH` → `MAT CH`), unterminated `/*` is kept verbatim and ends splitting so the backend surfaces the syntax error, and empty queries (trailing `;` + whitespace, or comment-only input) are filtered out.

File: web/src/__tests__/queries.test.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    const result = splitQueries('MATCH (p:Pod) /* tip: use ; sparingly */ RETURN p');
+    expect(result).toHaveLength(1);
+    expect(collapse(result[0])).toBe('MATCH (p:Pod) RETURN p');
+  });
+
+  test('multiple multi-line comments are each removed (no nesting)', () => {
+    const result = splitQueries('MATCH (p:Pod) /* a */ WHERE p.x = 1 /* b */ RETURN p');
+    expect(result).toHaveLength(1);
+    expect(collapse(result[0])).toBe('MATCH (p:Pod) WHERE p.x = 1 RETURN p');
+  });
```
[the remark this answers is not in the document]
Comment (naorpeled) [reply web/src/__tests__/queries.test.ts:47]: Addressed in 1938034. Added tests covering comment markers and `;` inside single- and double-quoted strings, token-boundary preservation, unterminated comments, trailing-`;`-with-whitespace producing no empty query, and comment-only input producing no queries.

**Comment by naorpeled** [comment]
@AvitalTamir the change grew a bit bigger than expected, let me know what you think

I'm also okay with skipping this for v1

File: pkg/core/lexer.go [hunk 3]
```diff
@@ -41,10 +41,11 @@ func (l *Lexer) NextToken() Token {
...
-		// Check for single-line comments (//)
+		// Check for comments (// single-line or /* */ multi-line)
 		if l.s.Peek() == '/' {
 			ch1 := l.s.Next() // Tentatively consume the first '/'
-			if l.s.Peek() == '/' {
+			switch l.s.Peek() {
+			case '/':
 				l.s.Next() // Consume the second '/'
 				// It's a comment, skip to the end of the line or EOF
 				for {
```
[the remark this answers is not in the document]
Comment (naorpeled) [reply pkg/core/lexer.go:83]: Addressed in 42f3dae. Confirmed the bug: `n/*c*/.x` produced ILLEGAL ("n/") and leaked the comment body as stray tokens. In the complex-identifier loop, a `/` that begins a comment (`//` or `/*`) now ends the identifier and the comment is skipped — so `n/*c*/.x` lexes as IDENT(n), DOT, IDENT(x). A `/` followed by a non-comment character is still a genuine separator, so annotation-style keys like `io/foo-bar` remain a single identifier (regression test added).

**Comment by AvitalTamir** [comment]
@naorpeled great job! I expected this to be a bigger change, no worries and awesome of you to take this.

I'll be comfortable with merging this if we cover this in the e2e suite.
A single test should be good (or more if there are edge cases? Can't think of any right now)

**Comment by naorpeled** [comment]
@AvitalTamir The E2Es are in :)

**Comment by naorpeled** [comment]
Also fixed Qodo's comments

**Review by AvitalTamir — APPROVED** [review]
